include/mips/syscall.h

```diff
@@ -27,8 +27,12 @@
   LEAF(name);                                                                  \
   li v0, num;                                                                  \
   syscall;                                                                     \
-  bnez v1, _C_LABEL(__sc_error);                                               \
+  bnez v1, ___sc_error##name;                                                  \
+  nop;                                                                         \
   j ra;                                                                        \
+  nop;                                                                         \
+  ___sc_error##name : j _C_LABEL(__sc_error);                                  \
```

Comment:
You're introducing lots of global symbols this way. Preferred way to code that is to use real local labels.
```
bnez v1, 1f
...
1:  j _C_LABEL(__sc_error)
```